### PR TITLE
Add remove/get job artifacts to cache endpoints.

### PR DESCRIPTION
**Pull request by fdc-ntflx**
### Context

Explain context and other details for this pull request.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java [hunk 1]
```diff
@@ -66,6 +66,10 @@ public interface ResourceCluster extends ResourceClusterGateway {
 
     CompletableFuture<Ack> addNewJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts);
 
+    CompletableFuture<Ack> removeJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts);
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java:69]: Do you need the clusterID?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java [hunk 2]
```diff
@@ -168,6 +168,8 @@ void storeAndUpdateWorkers(final IMantisWorkerMetadata existingWorker, final IMa
 
     void addNewJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts) throws IOException;
 
+    void removeJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts) throws IOException;
+
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java:172]: In order to make it simpler on the storage layer, could we just have a setter instead? what's the need to perform the delta/dedup on both the resource cluster layer as well as the storage layer?

**Review by sundargates — APPROVED** [review]
LGTM

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java (hunk 2 above)
Comment (fdc-ntflx) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java:172]: I don't know how simpler the implementation with a setter would look like. You'd still need a deleteAll on the pkey (aka clusterID) and addAll on remaining artifacts. I like the consistency of add and remove which will modify both RC and KV store.

**Pull request merged**